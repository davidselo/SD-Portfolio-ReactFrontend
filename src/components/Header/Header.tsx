import React from 'react';
import './Header.scss';
export const Header: React.FC = () => (
    <div className="header__topIntroduction">
        <div className='header__topIntroduction-text'>
            <p>Hello my name is David Villalba</p>
            <p>I am a full stack Developer</p>
        </div>
        <div className='light x1'></div>
        <div className='light x2'></div>
        <div className='light x3'></div>
        <div className='light x4'></div>
        <div className='light x5'></div>
        <div className='light x6'></div>
        <div className='light x7'></div>
        <div className='light x8'></div>
        <div className='light x9'></div>
    </div>
)
