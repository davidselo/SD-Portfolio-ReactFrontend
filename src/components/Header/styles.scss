@import '../../sass/styles.scss';

@media (min-width: $mobile) {
    .header--menu-items {
      display: none;
    }
  }
  @media (min-width: $tablet) {
    .header--menu-items {
      display: block;
    }
  }

