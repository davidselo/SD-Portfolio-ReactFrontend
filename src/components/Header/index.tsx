import React, {useState} from 'react';
import Link from '@mui/material/Link';
import {Link as RouterLink} from 'react-router-dom';

import {makeStyles} from '@material-ui/core/styles';
import AppBar from '@mui/material/AppBar';
import ToolBar from '@mui/material/Toolbar';
import Avatar from '@mui/material/Avatar';
import MenuIcon from '@mui/icons-material/Menu';
import SwipeableDrawer from '@mui/material/SwipeableDrawer';
import Divider from '@mui/material/Divider';
import ChevronRightIcon from '@mui/icons-material/ChevronRight';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import Grid from '@mui/material/Grid';
// import './styles.scss';
import Box from '@mui/material/Box';
import {NavigationLinks} from 'contracts/NavigationLinks';

// @todo: to get this data from Keystone
const navigationLinks: NavigationLinks = [
    {name: 'home', href: '/'},
    {name: 'cv', href: '/cv'},
    {name: 'Resume', href: '/resume'},
    {name: 'Contact', href: '/resume'},
];

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const useStyles = makeStyles(theme => ({
    link: {
        marginRight: 20,
    },
}));

const Header: React.FC = () => {
    const styles = useStyles();
    const [open, setOpen] = useState(false);
    /* istanbul ignore next */
    return (
        <AppBar position="sticky" color="default">
            <ToolBar disableGutters>
                <Grid container justifyContent="center">
                    <Grid item xl lg md sm={0} xs={0} />
                    <Grid item xl={6} lg={6} md={4} sm={9} xs={9}>
                        <Avatar>DS</Avatar>
                    </Grid>
                    <Grid item xl={4} lg={5} md={6} xs={2} textAlign="end">
                        <Box
                            component="div"
                            sx={{
                                display: {
                                    xl: 'block',
                                    lg: 'block',
                                    md: 'block',
                                    xs: 'none',
                                },
                                paddingTop: '7.25',
                            }}
                        >
                            <div className="header--menu-items">
                                {navigationLinks.map(item => (
                                    // eslint-disable-next-line react/jsx-key
                                    <Link
                                        className={styles.link}
                                        color="textPrimary"
                                        component={RouterLink}
                                        variant="button"
                                        underline="none"
                                        key={item.name}
                                        href={item.href}
                                        to={item.href}
                                    >
                                        {item.name}
                                    </Link>
                                ))}
                            </div>
                        </Box>
                    </Grid>
                    <Grid item xl lg={0} md sm xs textAlign="end">
                        <Box
                            component="div"
                            sx={{
                                display: {
                                    xl: 'none',
                                    lg: 'none',
                                    md: 'none',
                                    xs: 'block',
                                },
                            }}
                        >
                            <Box
                                className="header--burger-menu"
                                sx={{paddingTop: '8'}}
                                data-testid="header--burger-menu"
                            >
                                <MenuIcon
                                    onClick={() => setOpen(true)}
                                    data-testid="header--burger-menu-icon"
                                />
                            </Box>
                        </Box>
                    </Grid>
                </Grid>
            </ToolBar>

            <SwipeableDrawer
                open={open}
                anchor="right"
                onOpen={() => setOpen(true)}
                onClose={() => setOpen(false)}
                data-testid="header--swipeable-drawer"
            >
                <Box>
                    <ChevronRightIcon
                        onClick={() => setOpen(false)}
                        data-testid="header--chevron-icon"
                    />
                </Box>
                <Divider />
                {/* @todo: Create a function/helper to retrieve the links. DRY */}
                <List>
                    {navigationLinks.map(item => (
                        // eslint-disable-next-line react/jsx-key
                        <ListItem key={item.name}>
                            <Link
                                className={styles.link}
                                color="textPrimary"
                                component={RouterLink}
                                variant="button"
                                underline="none"
                                key={item?.name}
                                href={item.href}
                                to={item.href}
                            >
                                {item?.name}
                            </Link>
                        </ListItem>
                    ))}
                </List>
            </SwipeableDrawer>
        </AppBar>
    );
};

export default Header;
