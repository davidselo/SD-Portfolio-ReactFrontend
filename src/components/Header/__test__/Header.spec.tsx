import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react';
import Header from '../index';
import MenuIcon from '@mui/icons-material/Menu';
import ChevronRightIcon from '@mui/icons-material/ChevronRight';

describe('Header component', () => {
    test('Should match test snapshot', () => {
        const {container} = render(<Header />);
        expect(container).toMatchSnapshot();
    });
});
describe('MenuIcon component', () => {
    test('calls onClick prop when clicked', () => {
        const handleClick = jest.fn();
        render(
            <MenuIcon
                onClick={handleClick}
                data-testid="header--burger-menu-icon"
            />,
        );
        const element = screen.getByTestId('header--burger-menu-icon');
        fireEvent.click(element);
        expect(handleClick).toHaveBeenCalledTimes(1);
    });
});
describe('ChevronRightIcon component', () => {
    test('calls onClick prop when clicked', () => {
        const handleClick = jest.fn();
        render(
            <ChevronRightIcon
                onClick={handleClick}
                data-testid="header--chevron-icon"
            />,
        );
        const element = screen.getByTestId('header--chevron-icon');
        fireEvent.click(element);
        expect(handleClick).toHaveBeenCalledTimes(1);
    });
});
