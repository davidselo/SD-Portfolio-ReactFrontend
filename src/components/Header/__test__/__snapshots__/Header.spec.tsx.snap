// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should match test snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorDefault MuiAppBar-positionSticky css-1hnfq8z-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-true MuiGrid-grid-lg-true MuiGrid-grid-xl-true css-ehu4kr-MuiGrid-root"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 MuiGrid-grid-sm-9 MuiGrid-grid-md-4 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6 css-rwhae5-MuiGrid-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-md-6 MuiGrid-grid-lg-5 MuiGrid-grid-xl-4 css-14ysysj-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1ccvj77"
          >
            <div
              class="header--menu-items MuiBox-root css-0"
            >
              <a
                class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineNone css-m3nf4w-MuiTypography-root-MuiLink-root"
                href="/"
              >
                home
              </a>
              <a
                class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineNone css-m3nf4w-MuiTypography-root-MuiLink-root"
                href="/about-me"
              >
                about me
              </a>
              <a
                class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineNone css-m3nf4w-MuiTypography-root-MuiLink-root"
                href="/cv"
              >
                cv
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true MuiGrid-grid-sm-true MuiGrid-grid-md-true MuiGrid-grid-xl-true css-14xwpcs-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1vxiug2"
          >
            <div
              class="header--burger-menu MuiBox-root css-19qr23f"
              data-testid="header--burger-menu"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="header--burger-menu-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="PrivateSwipeArea-root css-x15wq9"
      style="width: 20px;"
    />
  </header>
</div>
`;
